src/frontend/src/handler/query.rs

```diff
@@ -79,19 +83,29 @@ pub fn gen_batch_query_plan(
         }
         (true, false) => QueryMode::Distributed,
         (false, true) => QueryMode::Local,
-        (false, false) => session.config().get_query_mode(),
+        (false, false) => match session.config().get_query_mode() {
+            QueryMode::Auto => determine_query_mode(batch_plan.clone()),
```

Comment:
When the mode is `Auto`, is there any way to show which mode is determined for the given query? (does `explain (trace)` help?)

Reply:
Yes, `explain (trace)` can tell you which mode is actually chosen.